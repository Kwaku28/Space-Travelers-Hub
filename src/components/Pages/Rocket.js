import Navbar from '../navbar';

const Rocket = () => (
  <Navbar />
);

export default Rocket;
