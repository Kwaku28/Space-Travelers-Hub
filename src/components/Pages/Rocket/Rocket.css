.container {
  border-top: 1px solid black;
  display: flex;
  padding-top: 20px;
  flex-direction: column;
  gap: 15px;
  margin: 5px 0;
}

.section {
  display: flex;
  flex-direction: row;
  font-family: sans-serif;
  gap: 10px;
}

.img {
  flex: 1;
}

.details {
  flex: 4;
  display: flex;
  flex-direction: column;
  gap: 10px;
}

.details p {
  padding-right: 5px;
}

.img img {
  width: 100%;
}
