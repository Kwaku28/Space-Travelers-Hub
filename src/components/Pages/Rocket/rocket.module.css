.reserve {
  background-color: rgb(255, 255, 255);
  color: #7a7676;
  width: 12vw;
}

.unreserve {
  background-color: rgb(144, 144, 226);
}

button {
  width: 10vw;
  height: 5vh;
  border: 0.1px solid black;
  color: #fff;
  border-radius: 3px;
  transition: 0.5s;
  font-size: 14px;
}

button:hover {
  background-color: rgb(28, 28, 71);
  color: #fff;
}

.badge {
  background-color: rgb(75, 181, 165);
  color: #fff;
  border-radius: 5px;
  margin: 0 3px;
  padding: 1px 3px;
}
