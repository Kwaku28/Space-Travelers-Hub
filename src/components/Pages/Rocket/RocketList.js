/*eslint-disable*/
import "./Rocket.css"
import FetchApi from "../../../Redux/rockets/Api";
import { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { toggleRocketStatus } from "../../../Redux/rockets/rocketsSlice";
import styles from './rocket.module.css';

export const RocketList = () => {
    const { error, rocketStatus, rockets } = useSelector((state) => state.rockets);
    const dispatch = useDispatch();
    useEffect(() => {
        dispatch(FetchApi());
    }, [dispatch]);
    
    if (error) {
        return "There is an error";
    }
    if (!rocketStatus) {
        return "...Loading";
    }

    return (
        <div className="container">
            {rockets.map((rocket) => {
                return (
                    <section className="section" key={rocket.id}>
                        <div className='img'>
                        <img src={rocket.RocketImages} alt="" />
                        </div>
                        <div className="details">
                            <h2>{rocket.RocketName}</h2>
                            <p> <span className={rocket.rocketStatus ? styles.badge : ''}>{rocket.rocketStatus? "Reseved" : ""}</span> {rocket.description}</p>
                            <button className={`${rocket.rocketStatus ? styles.reserve : styles.unreserve}`} onClick={() => dispatch(toggleRocketStatus(rocket.RocketId))}>
                            {rocket.rocketStatus ? 'Cancel reservation' : 'Reserve Rocket'}
                            </button>
                        </div>
                    </section>
                );
            })}
        </div>
    );
};
