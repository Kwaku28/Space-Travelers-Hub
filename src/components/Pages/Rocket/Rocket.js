/*eslint-disable*/

import Navbar from '../../navbar';
import { RocketList } from './RocketList';


const Rocket = () => (
  <>
  <Navbar />
  <RocketList />
  </>
  );

export default Rocket;
