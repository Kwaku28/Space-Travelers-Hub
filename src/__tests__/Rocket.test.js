import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rocket from '../components/Pages/Rocket/Rocket';
import store from '../Redux/store';

describe('Test Mission component', () => {
  it('Renders well', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Rocket />
      </Provider>,
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
