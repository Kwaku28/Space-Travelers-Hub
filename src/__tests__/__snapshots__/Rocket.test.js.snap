// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Mission component Renders well 1`] = `
<div
  className="container"
/>
`;
